### PR TITLE
Add GenericCollection::toArray()

### LINKED ISSUE #32: Convert to array
Some use cases require using plain PHP arrays, but getCollection() returns an object.

A naive conversion fails:

```php
$array = iterator_to_array(
    $manager->getCollection(...)->getIterator()
);
```

This fails because although $array is an array, its elements are objects of class `Aura\Marshal\Entity\GenericEntity`.

If this can be done with a custom builder, then a documentation example for that should be added.

### PR DESCRIPTION
**Pull request by compwright**
Resolves #32

### CODE REVIEW

**Comment by compwright** [comment]
@pmjones please be sure to review this!

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
         "php": "^7.2 || ^8.0"
     },
     "require-dev": {
-        "yoast/phpunit-polyfills": "^1.0.0"
+        "yoast/phpunit-polyfills": "^1.0.0",
+        "phpunit/phpunit": "^9.5"
```
Comment (harikt) [inline composer.json]: You don't need to add this. yoast/phpunit-polyfills will add phpunit according to the constraints.

File: composer.json [hunk 2]
```diff
@@ -26,5 +27,8 @@
         "psr-4": {
             "Aura\\Marshal\\": "tests/"
         }
+    },
+    "scripts": {
+        "test": "phpunit"
```
Comment (harikt) [inline composer.json]: It looks phpunit is global on your end. But if you really need this to work, you need to use "./vendor/bin/phpunit"

**Comment by harikt** [comment]
Thank you for the pull request. If you can make those small changes related to composer, we can merge.

**Review by koriym — APPROVED** [review]

**Pull request merged**